library/validation/validation_script.js.php

```diff
@@ -0,0 +1,134 @@
+<?php
+if($GLOBALS['new_validate']) {
+?>
+    <script type="text/javascript" src="<?php echo $GLOBALS['rootdir'] ?>/../library/js/vendors/moment.js"></script>
+    <script type="text/javascript" src="<?php echo $GLOBALS['rootdir'] ?>/../library/js/vendors/validate/validate.js"></script>
+    <script type="text/javascript" src="<?php echo $GLOBALS['rootdir'] ?>/../library/js/vendors/validate/validate_extend.js"></script>
+<?php
+ }
+?>
+
+<script language='JavaScript'>
+
+    function submitme(new_validate,e,form_id) {
+
+    //Use the old validation script if no parameter sent (backward compatibility)
+        if (new_validate !== 1) {
+            var f = document.forms[0];
+            if (validate(f)) {
+                top.restoreSession();
+                f.submit();
+            } else {
+                e.preventDefault();
+            }
+        } else {
+
+            <?php
+
+            $constraints = LBF_Validation::generate_validate_constraints($form_id);
+
+            ?>
+            var valid = true ;
+            var constraints = <?=$constraints?>;
+            var error_msg ='<?=xl('is not valid')?>';
+            var form = document.querySelector("form#"+form_id);
+            var elements = validate.collectFormValues(form);
```

Comment:
This is confusing.  Looks like there is a validate object as well as the caller's existing validate() function.  It's not obvious what's being referenced here.